playerctl/playerctl-daemon.c

```diff
@@ -752,22 +811,49 @@ static void player_signal_proxy_callback(GDBusConnection *connection, const gcha
         }
     }
 
-    g_dbus_connection_emit_signal(ctx->connection, NULL, object_path, interface_name, signal_name,
-                                  parameters, &error);
-    if (error != NULL) {
-        g_debug("could not emit signal: %s", error->message);
-        g_clear_error(&error);
+    if (is_properties_changed || player == context_get_active_player(ctx)) {
+        g_dbus_connection_emit_signal(ctx->connection, NULL, object_path, interface_name,
+                                      signal_name, parameters, &error);
+        if (error != NULL) {
+            g_debug("could not emit signal: %s", error->message);
+            g_clear_error(&error);
+        }
     }
 }
 
 int main(int argc, char *argv[]) {
     struct PlayerctldContext ctx = {0};
     GError *error = NULL;
+
+    if (argc > 2 || (argc == 2 && strcmp(argv[1], "--shift") != 0)) {
```

Comment:
It would be better to use the glib argument parser for this. See the playerctl source code. Someone will have to do this at some point.

Reply:
I tried my best on the glib parser. It's a little weird since I'm using `G_OPTION_REMAINING` to catch the single parameter that we are interested in (i.e. `shift`), but I don't think there it is possible to indicate that it is optional (unless we make it a flag, `--shift`, which you don't want).